x-pack/plugins/security_solution/public/detections/components/rules/rule_switch/index.tsx

```diff
@@ -72,6 +76,10 @@ export const RuleSwitchComponent = ({
     [enabled, executeBulkAction, id, onChange, startMlJobsIfNeeded, startTransaction]
   );
 
+  useEffect(() => {
+    setAriaLabel(ruleName ? ruleSwitchAriaLabel(ruleName, enabled) : undefined);
+  }, [enabled, ruleName]);
```

Comment:
I think it would be better to do it using `useMemo`:

```suggestion
  const arialLabel = useMemo((): string | undefined => {
    return ruleName ? ruleSwitchAriaLabel(ruleName, enabled) : undefined;
  }, [ruleName, enabled]);
```

This common pattern of updating a component's state based on its props or state via `useEffect` is described in https://react.dev/learn/you-might-not-need-an-effect and the recommendation is to avoid that to not trigger extra rendering cycles.

Reply:
I think in accordance with  doc above it simple calculation which we should be not wrapped